### ISSUE TITLE
Permission option for specific block/entity protection(s)

### ISSUE THREAD

**Issue by MC-MrBirdy** [pop4959/Bolt#62]
A suggestion for a feature to add the option to configure permissions for specific blocks/entities to only allow protecting them when having said permission.

For example:
```
blocks:
  chest:
    autoProtect: private
  lever:
    autoProtect: false
    permission: bolt.block.moderator
  stone_button:
    autoProtect: false
    permission: bolt.block.moderator
  spawner:
    autoProtect: false
    permission: bolt.block.admin
```

That way the permission "bolt.block.moderator" or "bolt.block.admin" can be given to specific players/groups.
Allowing for example only moderators to protect a lever and stone_button.

When no permission is specified all players can use it, which would be the default.

Being able to define the permission name can allow for multiple blocks/entities being covered by the same permission.

**Comment by pop4959** [comment]
I think it might be better to have a configuration option that enables this permission for all blocks and assigns a specific permission for them. I don't think it makes that much sense to make the actual node configurable, since grouping nodes can easily be done via a permission plugin anyway (which you will have to do per-permission/block anyways).

Let me know what you think.

**Comment by MC-MrBirdy** [comment]
That's a suitable option as well, and indeed permissions, within LuckPerms at least, can be grouped.

**Comment by pop4959** [comment]
Implemented in 29f40feb65d3812b07a80dc983280f6876b4644c.

How it works:
- Each protectable type gets its own `autoProtectPermission` and `lockPermission` (by default, these are not included in the configuration, you need to add them).
- If you enable the permission, you will need the `bolt.protection.autoprotect.<type>` or `bolt.protection.lock.<type>` for the protectable type (which is the same name it is under in the config).
